src/Client.ts

```diff
@@ -14,34 +14,47 @@ interface Client {
 }
 
 class Client extends EventEmitter {
-  private ws = new WebsocketShard(this.token, this);
+  private ws = new WebsocketShard(this);
   public gatewayURL: string = "";
-  public options: ClientOptions;
+  public requestHandler = new RequestHandler(this);
 
-  // TODO(Z): This may have implications on boot times.
   public constructor(
-    private token = Deno.env.get("TOKEN") || "",
-    options?: ClientOptions,
+    public token: string,
+    public options: ClientOptions = {
+      compress: false,
+      compressStream: true,
+      intents: undefined,
+    },
   ) {
     super();
-    this.options = {
-      compress: false,
-      ...(options || {}),
-    };
+    if (!token.startsWith("Bot ")) this.token = `Bot ${this.token}`;
   }
 
   public async connect(): Promise<void> {
-    // TODO: implement proper ratelimiting support
-
     if (!this.gatewayURL) {
-      const { url } = await fetch(`${API_BASE}/gateway`).then((r) => r.json());
+      const { url, session_start_limit: ssl } = await this.requestHandler
+        .request("GET", `/gateway/bot`);
+      if (!ssl.remaining) {
+        throw new Error(
+          `Starting the bot would reset the token. Please restart the bot after ${ssl.reset_after}ms`,
+        );
+      }
       this.gatewayURL = `${url}?v=6&encoding=json${
         this.options.compressStream ? `&compress=zlib-stream` : ""
       }`;
     }
     return this.ws.connect();
   }
 
+  public createMessage(channelID: string, content: any): Promise<object> {
+    return <Promise<object>> this.requestHandler.request(
```

Comment:
You don't need to do this, the return type is inferred by the type assignment.

Reply:
it's inferred as `Promise<any>`, so yeah no

Reply:
oh, like above, I prefer explicit typing

Reply:
Problem is though, they both speak the same message.